### PR TITLE
replace toml templates with toml-rb gem

### LINKED ISSUE #32: Cannot differentiate string and numeric options
I am using the following Puppet code:

```
  telegraf::input { 'statsd':
    options => {
      service_address          => '127.0.0.1:8125',
      allowed_pending_messages => 10000,
    },
  }
```

This results in the following Telegraf configuration:

```
[[inputs.statsd]]
  allowed_pending_messages = "10000"
  service_address = "127.0.0.1:8125"
  templates = ["logstash.* namespace.sender.measurement*"]
```

This configuration is invalid and Telegraf reports the following error:

```
Error parsing /etc/telegraf/telegraf.d/statsd.conf, line 2: statsd.Statsd.AllowedPendingMessages: `string' type is not assignable to `int' type
```

The problem here is that Puppet doesn't differentiate between numbers and strings (at least until Puppet 4, unless using the future parser). I was able to workaround this issue with the following Puppet code:

```
  telegraf::input { 'statsd':
    options => {
      service_address          => '127.0.0.1:8125',
      allowed_pending_messages => 0 + 10000,
    },
  }
```

Ideally though, the template used by `telegraf::input` would check if a configuration option is numeric and remove the quotes.

### PR DESCRIPTION
**Pull request by nrdmn**
This uses a real toml generator instead of templates.
It is an alternative approach to  #47 and #79 and fixes #32, #38, #41, #42 and #68.

Tests are still mostly broken because I don't know ruby.

### CODE REVIEW

**Comment by yankcrime** [comment]
Thanks for this @nrdmn - this looks like the most sensible way forward.  I'll see about fixing those tests, and then this would be ideal for v2.0 of the module.

**Comment by hanej** [comment]
There's some functionality that we require (like having multiple inputs of the same type in hiera) that I am working on adding.  It looks like this PR solves that and much more.  Looking forward to seeing this in a new version.

**Comment by hanej** [comment]
@nrdmn I created a PR for your branch that may fix the spec tests and also added plugin_type through hiera so we can have the same input/output multiple times.

**Comment by yankcrime** [comment]
@hanej @nrdmn Any progress with filling the gaps in functionality?  I'd like to get this merged ASAP if at all possible as it fixes a lot of issues.

**Comment by hanej** [comment]
The ball is in @nrdmn's court.  I issued a pull request to his repo.

**Comment by yankcrime** [comment]
Thanks both!

**Review by yankcrime — APPROVED** [review]

File: spec/acceptance/default_spec.rb [hunk 1]
```diff
@@ -12,18 +12,18 @@
...
                     'urls'     => [ 'http://localhost:8086' ],
                     'database' => 'telegraf',
                     'username' => 'telegraf',
                     'password' => 'metricsmetricsmetrics',
-                  }
+                  }]
               },
               inputs    => {
-                  'cpu' => {
+                  'cpu' => [{
```
Comment (sparr) [inline spec/acceptance/default_spec.rb:23]: This same change needs to be made in manifests/params.pp defaults. (sorry for being so late with this comment) @hanej @nrdmn

**Pull request merged**